### ISSUE TITLE
Generate auto-incrementing asset IDs: populate asset tag on save instead of creation of asset

### ISSUE THREAD

**Issue by n-v-o** [snipe/snipe-it#6041]
**Server (please complete the following information):**
 - Snipe-IT Version : 4.6.3
 - OS: Windows Server 2012 R2
 - Web Server: IIS
 - PHP Version: 7.2.2

**Is your feature request related to a problem? Please describe.**
Yes it is. We have several people who create assets for our team. If two (or more) people open up the asset form before an asset is saved, they both get the same asset tag number. This results in one person being able to save, and the other person having an error that the asset tag needs to be unique.

**Describe the solution you'd like**
When auto-increment is enabled, The asset tag can be generated on save. After hitting save, a query could be made to see what asset tag is next in line and use that. If it fails, it could try again but ultimately go back to the web form asking to try again with some context on what happened.

Some people will still want the ability to edit manually, which I think would be fine by editing the asset afterwards.

**Describe alternatives you've considered**

- Manually assigning asset tags, which is cumbersome/annoying.
- Manually increasing/changing the value, but the auto-increment feature doesn't seem to catch these so this causes problems, and a super user has to adjust the next available. E.G. If we both have the asset tag of 0005, I could make it 0006 manually, but the next auto-increment field is not updated to reflect that.

This is semi-related to #4577, but is different.

**Comment by snipe** [comment]
Well, the problem there is that if you expected a certain asset tag when you started the form, it won’t necessarily be the asset tag you actually get, which could be super confusing to people who are expecting that tag to be a source of truth when they’re creating the asset.

**Comment by n-v-o** [comment]
I understand where you are coming from. In our situation we have 13 people who add assets, and any number of them may be adding an asset at the same time. We are going to run into this issue a lot. (We moved to production a couple days ago, and this has happened 4-5 times already in one day). From one perspective, if someone has that option enabled, they aren't looking to edit asset tag numbers unless absolutely necessary, but I could be wrong there. We are certainly looking for a hands off approach. For our use case I'd prefer seeing a place holder for the field that said "asset tag generated on save", then if I needed to change it for some reason we could go in and edit the asset. Does that make sense?
